pkg/api/signing_cert.go

```diff
@@ -17,66 +17,74 @@ limitations under the License.
 package api
 
 import (
-	"encoding/base64"
+	"crypto/ecdsa"
+	"crypto/x509"
 	"encoding/pem"
+	"errors"
+	"fmt"
 	"net/http"
+	"strings"
 
+	"github.com/coreos/go-oidc/v3/oidc"
 	"github.com/go-openapi/runtime/middleware"
 	fca "github.com/sigstore/fulcio/pkg/ca"
 	"github.com/sigstore/fulcio/pkg/ctl"
-	"github.com/sigstore/fulcio/pkg/generated/models"
 	"github.com/sigstore/fulcio/pkg/generated/restapi/operations"
 	"github.com/sigstore/fulcio/pkg/log"
+	"github.com/sigstore/fulcio/pkg/oauthflow"
 	"github.com/spf13/viper"
-	"golang.org/x/net/context"
 )
 
-func SigningCertHandler(params operations.SigningCertParams, principal interface{}) middleware.Responder {
+func SigningCertHandler(params operations.SigningCertParams, principal *oidc.IDToken) middleware.Responder {
+	ctx := params.HTTPRequest.Context()
+	emailAddress, _, _ := oauthflow.EmailFromIDToken(principal)
```

Comment:
do we need to handle errors here?

Reply:
In theory, no. This code path should only be called after the request was authenticated, which includes validation of the ID token.

Reply:
added code in next commit to be defensive here